docs/samples/browser-plugin-meetings/app.js

```diff
@@ -350,7 +359,85 @@ function createMeeting(e) {
   return false;
 }
 
-function joinMeeting({withMedia, withDevice} = {withMedia: false, withDevice: false}) {
+function refreshCaptcha() {
+  const meeting = webex.meetings.getAllMeetings()[selectedMeetingId];
+  meeting.refreshCaptcha()
+    .then(() => {
+      console.log('MeetingsManagement#refreshCaptcha() :: successfully refreshed captcha');
+      meetingsJoinCaptchaImgElm.src = meeting.requiredCaptcha.verificationImageURL;
+      meetingsJoinCaptchaImgElm.hidden = false;
+      meetingsJoinCaptchaElm.type = "text";
+      meetingsJoinCaptchaElm.value = "";
+      refreshCaptchaElm.hidden = false;
+    })
+    .catch((error) => {
+      console.error('MeetingsManagement#refreshCaptcha() :: error refreshing captcha', error);
+      throw (error);
+    });
+}
+
+meetingsListElm.onclick = (e) => {
+  selectedMeetingId = e.target.value;
+  const meeting = webex.meetings.getAllMeetings()[selectedMeetingId];
+
+  if(meeting && meeting.passwordStatus === 'REQUIRED'){
+    meetingsJoinPinElm.disabled = false;
+    verifyPasswordElm.disabled = false;
+    document.getElementById('btn-join').disabled = true;
+    document.getElementById('btn-join-media').disabled = true;
+  } else {
+    meetingsJoinPinElm.disabled = true;
+    verifyPasswordElm.disabled = true;
+    document.getElementById('btn-join').disabled = false;
+    document.getElementById('btn-join-media').disabled = false;
+  }
+}
+
+function verifyPassword() {
+  const meeting = webex.meetings.getAllMeetings()[selectedMeetingId];
+
+  if (!meeting) {
+    throw new Error(`meeting ${selectedMeetingId} is invalid or no longer exists`);
+  }
+
+  const joinOptions = {
+    pin: meetingsJoinPinElm.value,
+    captcha: meetingsJoinCaptchaElm.value
+  };
+
+  if (meeting && meeting.passwordStatus === 'REQUIRED') {
+    meeting
+      .verifyPassword(joinOptions.pin, joinOptions.captcha)
+      .then((res) => {
+        if (res.failureReason === 'NONE') {
+          passwordCaptchaStatusElm.innerText = 'Password is verified';
+          passwordCaptchaStatusElm.style.backgroundColor = '#49e849';
+          verifyPasswordElm.disabled = true;
+          document.getElementById('btn-join').disabled = false;
+          document.getElementById('btn-join-media').disabled = false;
+          // joinMeetingNow();
+        } else if (res.failureReason === "WRONG_PASSWORD") {
+          passwordCaptchaStatusElm.innerText = 'Password is required';
+          passwordCaptchaStatusElm.style.backgroundColor = '#fa6e6e';
+        } else if (res.failureReason === 'WRONG_CAPTCHA') {
+          passwordCaptchaStatusElm.innerText = 'Password & Captcha is required';
```

Comment:
theoretically it is possible for wbxappapi to say that only captcha is required (without password), so maybe it would be better to add handling for that. Also probably better instead of checking the `failureReason` to check the  `isPasswordValid` and `requiredCaptcha` fields

Reply:
Let me check the isPasswordValid and requiredCaptcha fields. Thanks for pointing out.